photutils/detection/findstars.py

```diff
@@ -140,37 +106,126 @@ def daofind(data, threshold, fwhm, ratio=1.0, theta=0.0, sigma_radius=1.5,
-    tbl.add_column(idcol, 0)
-    if len(tbl) == 0:
-        warnings.warn('Sources were found, but none pass the sharpness and '
-                      'roundness criteria.', AstropyUserWarning)
-    return tbl
-
+    def __init__(self, threshold, fwhm, ratio=1.0, theta=0.0,
+                 sigma_radius=1.5, sharplo=0.2, sharphi=1.0, roundlo=-1.0,
+                 roundhi=1.0, sky=0.0, exclude_border=False):
+        """
+        Parameters
+        ----------
+        threshold : float
+            The absolute image value above which to select sources.
+        fwhm : float
+            The full-width half-maximum (FWHM) of the major axis of the
+            Gaussian kernel in units of pixels.
+        ratio : float, optional
+            The ratio of the minor to major axis standard deviations of the
+            Gaussian kernel.  ``ratio`` must be strictly positive and less
+            than or equal to 1.0.  The default is 1.0 (i.e., a circular
+            Gaussian kernel).
+        theta : float, optional
+            The position angle (in degrees) of the major axis of the
+            Gaussian kernel measured counter-clockwise from the positive x
+            axis.
+        sigma_radius : float, optional
+            The truncation radius of the Gaussian kernel in units of sigma
+            (standard deviation) [``1 sigma = FWHM /
+            (2.0*sqrt(2.0*log(2.0)))``].
+        sharplo : float, optional
+            The lower bound on sharpness for object detection.
+        sharphi : float, optional
+            The upper bound on sharpness for object detection.
+        roundlo : float, optional
+            The lower bound on roundess for object detection.
+        roundhi : float, optional
+            The upper bound on roundess for object detection.
+        sky : float, optional
+            The background sky level of the image.  Setting ``sky`` affects
+            only the output values of the object ``peak``, ``flux``, and
+            ``mag`` values.  The default is 0.0, which should be used to
+            replicate the results from `DAOFIND`_.
+        exclude_border : bool, optional
+            Set to `True` to exclude sources found within half the size of
+            the convolution kernel from the image borders.  The default is
+            `False`, which is the mode used by `DAOFIND`_.
+        """
 
-def irafstarfind(data, threshold, fwhm, sigma_radius=1.5, minsep_fwhm=2.5,
-                 sharplo=0.5, sharphi=2.0, roundlo=0.0, roundhi=0.2,
-                 sky=None, exclude_border=False):
+        self.threshold = threshold
+        self.fwhm = fwhm
+        self.ratio = ratio
+        self.theta = theta
+        self.sigma_radius = sigma_radius
+        self.sharplo = sharplo
+        self.sharphi = sharphi
+        self.roundlo = roundlo
+        self.roundhi = roundhi
+        self.sky = sky
+        self.exclude_border = exclude_border
+
+    def __call__(self, data):
+        """
```

Comment:
Similarly, this docstring should probably be moved to the `find_stars` method to make sure it's included in the rendered docs.

You can always `python setup.py build_sphinx` locally to check that that docs look like you expect them to look and include all relevant pieces.